integrations/integrations.go

```diff
@@ -0,0 +1,201 @@
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/server"
+	serverconfig "github.com/cosmos/cosmos-sdk/server/config"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	"github.com/polymerdao/monomer"
+	"github.com/polymerdao/monomer/environment"
+	"github.com/polymerdao/monomer/genesis"
+	"github.com/polymerdao/monomer/node"
+	"github.com/polymerdao/monomer/testapp"
+)
+
+func StartCommandHandler(
+	svrCtx *server.Context,
+	clientCtx client.Context,
+	appCreator servertypes.AppCreator,
+	inProcessConsesus bool,
+	opts server.StartCmdOptions,
+) error {
+	// We assume `inProcessConsensus` is true for now, so let's return an error if it's not.
+	if !inProcessConsesus {
+		return fmt.Errorf("in-process consensus must be enabled")
+	}
+
+	// --- Do the normal Cosmos SDK stuff ---
+	svrCfg, err := GetAndValidateConfig(svrCtx)
+	if err != nil {
+		return fmt.Errorf("failed to get and validate server config: %w", err)
+	}
+
+	app, appCleanupFn, err := startApp(svrCtx, appCreator, opts)
+	if err != nil {
+		return fmt.Errorf("failed to start application: %w", err)
+	}
+	defer appCleanupFn()
+
+	// Would usually start a Comet node in-process here, but we replace the
+	// Comet node with a Monomer node.
+	return startInProcess(svrCtx, svrCfg, clientCtx, app, opts)
+}
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L624
+func startApp(
+	svrCtx *server.Context,
+	appCreator servertypes.AppCreator,
+	opts server.StartCmdOptions,
+) (app servertypes.Application, cleanupFn func(), err error) {
+	traceWriter, traceCleanupFn, err := SetupTraceWriter(svrCtx.Logger, svrCtx.Viper.GetString(FlagTraceStore))
+	if err != nil {
+		return app, cleanupFn, err
+	}
+
+	home := svrCtx.Config.RootDir
+	db, err := opts.DBOpener(home, server.GetAppDBBackend(svrCtx.Viper))
+	if err != nil {
+		return app, cleanupFn, err
+	}
+
+	// TODO: Check if is testnet and implement `testnetify` function
+
+	app = appCreator(svrCtx.Logger, db, traceWriter, svrCtx.Viper)
+	cleanupFn = func() {
+		traceCleanupFn()
+		if localErr := app.Close(); localErr != nil {
+			svrCtx.Logger.Error(localErr.Error())
+		}
+	}
+
+	return app, cleanupFn, nil
+}
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L307
+// We modify this function to start a Monomer node in-process instead of a Comet node.
+func startInProcess(
+	svrCtx *server.Context,
+	svrCfg serverconfig.Config,
+	clientCtx client.Context,
+	app servertypes.Application,
+	opts server.StartCmdOptions,
+) error {
+	cmtCfg := svrCtx.Config
+	gRPCOnly := svrCtx.Viper.GetBool(FlagGRPCOnly)
+
+	g, ctx := GetCtx(svrCtx, true)
+
+	if gRPCOnly {
+		svrCtx.Logger.Info("starting node in gRPC only mode; Monomer is disabled")
+		svrCfg.GRPC.Enable = true
+	} else {
+		svrCtx.Logger.Info("starting Monomer node in-process")
+		// Start the Monomer node
+		_, monomerCleanupFn, err := startMonomerNode()
```

Comment:
We can pass in an `environment.Env` here to avoid dealing with the awkward cleanup function.

Reply:
Addressed.